transitions/core.py

```diff
@@ -529,6 +530,8 @@ def add_transition(self, trigger, source, dest, conditions=None,
             after = listify(after) + listify(self.after_state_change)
 
         for s in source:
+            if isinstance(dest, State):
+                dest = dest.name
```

Comment:
Wait, no--that's checking `dest`, not `source`. At this point `dest` hasn't been converted, only `source`.